### ISSUE TITLE
how to preprocess the self.x in graph

### ISSUE THREAD

**Issue by ai-galgame** [Kyubyong/transformer#126]
Hello, Kyubyong. Nice to meet you.
It's so kind of you for publishing such a good implementation of the Transformer.
I have learned a lot from your project.
Recently, I'm doing a test by modifying your source code, but I encountered a problem.
If you are available, please help me, thank you.
___
In my task, I would like to change the input_x and output_y not with the word-level, but with the sentence-level. 
I wish to change the elements of input_x from words to sentences, and use a [bert-as-service](https://github.com/hanxiao/bert-as-service/) tool to make sentence-level embedding as encoder's input. 
Because of the input of bert-as-service needs to be the type of string list, (i.e.['First do it', 'then do it right', 'then do it better']), so I changed the type of input_x from int vocab_id to string of sentences, like this:
```
def get_batch_data():    
    X, Y = load_train_data() 
    X = tf.convert_to_tensor(X, tf.string)  #[,30]
    Y = tf.convert_to_tensor(Y, tf.int32)   #[,30]
    input_queues = tf.train.slice_input_producer([X, Y])
    x, y = tf.train.shuffle_batch(input_queues,...)
    return x, y  #[batch_size,30]
```
X becomes like this:
```
[['Some of them were not sure how to use it.',
  'They thought it was a vegetable and tried cooking the leaves.',
  ...'[PAD]','[PAD]','[PAD]']
 ['One day, he came to visit a saint and wanted to be his student.'
  'The saint provided some tea.'
  ...'[PAD]','[PAD]','[PAD]']...]
```
And then, in modules.py, I defined a function to call bert-as-service for encoding sentences, just like the word_embedding:
```
def encoding_sents_by_bert(inputs):  #inputs: self.x.eval() from Graph.graph
    bc = BertClient()
    outputs_vec = np.empty(shape=[0, hp.hidden_units])
    for ipt in inputs:
        n = []
        for i in ipt.tolist():
            n.append(str(i))
        outputs_vec = np.append(outputs_vec, bc.encode(n), axis = 0)
    outputs_vec = outputs_vec.reshape([-1,30,hp.hidden_units])
    print(np.shape(outputs_vec))  #[batch,30,768]
    opt_vec = tf.convert_to_tensor(outputs_vec, tf.float32)
    return opt_vec
```
The result of sentence encoding will be converted to tensor and return.
The problem is in the training.
As the tf.train.shuffle_batch can only return tensor type, so I need to run the "self.x.eval()" to get the content of batch_x in Graph:
```
class Graph():
    def __init__(self, is_training=True):
        self.graph = tf.Graph()
        with self.graph.as_default():
            if is_training:
                self.x, self.y = get_batch_data()  #[batch,30]
            # Encoder
            with tf.variable_scope("encoder"):
                ## Embedding
                self.sess = tf.InteractiveSession()
                coord = tf.train.Coordinator()
                threads = tf.train.start_queue_runners(coord=coord)
                self.enc_s = encoding_sents_by_bert(self.x.eval())  #[batch,30,768]
                coord.request_stop()
                coord.join(threads)
                self.sess.close()
                self.enc = self.enc_s
                '''
                self.enc_s = embedding(self.x, vocab_size, num_units, scope="enc_embed")
                self.enc = self.enc_s
                '''
#Training
if __name__ == '__main__': 
    g = Graph("train"); print("Graph loaded")
    sv = tf.train.Supervisor(graph=g.graph, ...)
    with sv.managed_session(config=configGPU) as sess:
        for epoch in range(1, hp.num_epochs+1): 
            if sv.should_stop(): break            
            for step in range(g.num_batch):
                sess.run(g.train_op)
                print(sess.run(g.x)[0])  #used for test
                print(sess.run(g.enc_s)[0])  #used for test
```
I set a tf.InteractiveSession() to run x.eval(), and use the "encoding_sents_by_bert" to instead the original word_embedding function to get the sentence-embedding as self.enc.
But it can only be executed when creating an instance of the Graph ( g = Graph("train") ), the console of bert-as-service could show requests' records and the self.enc_s also could receive values.
When in training step (sess.run(g.train_op)), the self.enc_s never changed.

Through the _print(sess.run(g.x)[0])_, I could see the self.x changed every time, but the _print(sess.run(g.enc_s)[0])_ output a totally same value in every time. And the console of bert-as-service also has no responses.
More weird is, even the self.enc_s never changed, the training step is continuously running! It seems like using the self.enc_s which from the first initialization and changing the self.enc every time.

Sorry for troubling you, because I'm a beginner of the TensorFlow and this problem really confused me a lot.
If you know how to correct it, please tell me.
Thank you very much!
